CruzHacks/cruzhacks-2021-website
# Adding Email Subscription Component

Problem
=======
As a user,
I would like to subscribe to CruzHacks' newsletter
to receive timely updates about this years hackathon!

[Link to Asana Ticket](https://app.asana.com/0/1197738625416321/1197738625416326/f)

Solution
========
To create an email subscription component, we first create a responsive element based on the provided Figma design, which only includes an input field and submission button. The two elements are separated by a single right margin on `input` and wraps left-aligned using `flex-wrap: wrap`. The `input` component itself then resizes according to the width of the current viewports.

To handle user input, the `input`'s `onCapture` event is handled, which updates a local state variable corresponding to the user's current input. We also capture its `keyDown` event to submit their email whenever they input the "Enter" key. On the `button` component, we then submit their email when the button is clicked.

To create an asynchronous request, Axios is used to send an HTTP request to the designated Mailchimp endpoint and a feedback message is appropriately bound and displayed to the user using the `showFeedback` conditional.

Change summary:
---------------
* Added more accurate `Arrow` SVG and updated minimum breakpoint to include iPhone X 475c3fd0575617db4287fd9542c4f65d1387fa01
* Finished responsive layout e9ed21da4223269e0b97f8396e712bd1ace53067
* Handled user input  e31bb975dc125051302aa380fe688c7ef09e4c7e
* Configured Axios request 94c3d81be87ff24c139329ee9c02820c72cb6d9e

Steps to Verify:
----------------
1. Checkout branch using `git checkout feature/email-subscription`
2. Start local development server using `yarn start`
3. Open Chrome Developer Tools using `CMD/CTRL + SHIFT + I` to verify that email subscription component is responsive
4. Submit email in form to verify that appropriate response message is returned
